src/SM/StateMachine/StateMachine.php

```diff
@@ -212,20 +213,23 @@ protected function setState($state)
      * Builds and calls the defined callbacks
      *
      * @param TransitionEvent $event
-     * @param string          $position
+     * @param string $position
+     * @return bool
      */
     protected function callCallbacks(TransitionEvent $event, $position)
     {
+        $result = true;
         if (!isset($this->config['callbacks'][$position])) {
-            return;
+            return $result;
         }
 
         foreach ($this->config['callbacks'][$position] as &$callback) {
             if (!$callback instanceof CallbackInterface) {
                 $callback = $this->callbackFactory->get($callback);
             }
 
-            call_user_func($callback, $event);
+            $result = $result & call_user_func($callback, $event);
```

Comment:
Indeed you're right! Thanks for having switched the operands.
It means that as per your implementation, if the first guard prevents the transition, all the next guards will also be executed. Not sure about this behaviour, what do you think?
I agree that we want all `before` and `after` callbacks to be executed even if one of them returns `false`, but do we want the same for guards?